docs/basic-usage.md

```diff
@@ -0,0 +1,88 @@
+# Basic Usage
+
+The following is a basic annotated example of how to use this crate.
+
+## Interfaces
+
+Let's assume that you want to interact with a COM API from Rust. This COM API already exists and is documented through the use of an IDL declaration.
+
+```
+typedef struct Data
+{
+    float x;
+    float y;
+} Data;
+
+interface IMyInterface : IUnknown
+{
+    HRESULT MyMethod(
+        [in] const Data *my_data,
+        [in] long n,
+        [out] IMyOtherInterface **other
+    );
+}
+
+interface IMyOtherInterface : IMyInterface
+{
+    HRESULT MyOtherMethod();
+}
+```
+
+We must translate this IDL direclty into Rust code using `com::interfaces!` macro.
+
+Take note that each method and argument must be in the same exact order and must have the same in-memory properties as the types declared in the IDL.
+
+```rust 
+/// The Data struct must match the in-memory layout exactly. We therefore use `#[repr(C)]`.
+#[repr(C)]
+struct Data {
+    x: f32, // IDL's float type and Rust's f32 have the same in-memory layout
+    y: f32, // Note, the order of the fields here matters not necessarily their names
+}
+
+com::interfaces! {
+    unsafe interface IMyInterface: IUnknown {
+        fn MyMethod(
+            my_data: &Data,
+            n: i64, // C's long is equivalent to Rust's i64
```

Comment:
That's compiler dependent. C/C++'s `long` is typically either 32 bits (Visual C++) or 64 bits (most Clang implementations). You should just avoid `long` in ABI declarations.